applications/portal/backend/api/views/rest/experiment.py

```diff
@@ -124,16 +129,61 @@ def upload_files(self, request, **kwargs):
         except InvalidInputError:
             return Response(status=status.HTTP_400_BAD_REQUEST)
         try:
-            dir_path = create_temp_dir(CORDMAP_DATA)
-            filepaths = move_files([key_file, data_file], dir_path)
+            filepaths = move_files([key_file, data_file], instance.storage_path)
         except Exception as e:
             return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
         try:
-            created = upload_files(instance, filepaths[KEY_INDEX], filepaths[DATA_INDEX])
+            created = upload_pair_files(instance, filepaths[KEY_INDEX], filepaths[DATA_INDEX])
             response_status = status.HTTP_201_CREATED if created else status.HTTP_200_OK
         except InvalidInputError:
             response_status = status.HTTP_400_BAD_REQUEST
         return Response(status=response_status)
 
+    @action(
+        detail=True,
+        methods=["post"],
+        parser_classes=(MultiPartParser,),
+        name="experiment-upload-single-file",
+        url_path="upload-single-file",
+    )
+    def upload_single_file(self, request, **kwargs):
+        instance = self.get_object()
+        file = request.FILES.get("file")
+        if file is None:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        try:
+            filepaths = move_files([file], instance.storage_path)
+        except Exception as e:
+            return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        try:
+            created = upload_single_file(instance, filepaths[0])
+            response_status = status.HTTP_201_CREATED if created else status.HTTP_200_OK
+        except InvalidInputError:
+            response_status = status.HTTP_400_BAD_REQUEST
+        return Response(status=response_status)
+
+    @action(detail=True, methods=['get'],  url_path='download_populations/(?P<active_populations>[^/.]+)')
+    def download_populations(self, request, pk=None, active_populations=None):
+        experiment = self.get_object()
+
+        active_populations = active_populations.split(',')
```

Comment:
@afonsobspinto I don't like the string parameter and splitting it here into an array. Why not use an array as the parameter type